addons/gdUnit3/src/update/GdUnitUpdate.gd

```diff
@@ -128,14 +128,20 @@ func _on_update_pressed():
 	var plugin := EditorPlugin.new()
 	# close gdUnit scripts before update
 	close_open_editor_scripts(plugin)
-
+	
 	plugin.get_editor_interface().set_plugin_enabled("gdUnit3", false)
 	
 	# extract zip to tmp
-	update_progress("extracting zip ..")
 	var source := ProjectSettings.globalize_path(zip_file)
 	var dest := ProjectSettings.globalize_path(tmp_path)
-	OS.execute("tar", ["-xf", source, "-C", dest])
+	
+	update_progress("extracting zip '%s' to '%s'" % [source, dest])
+	var result := _extract_package(source, dest)
+	if result.is_error():
+		update_progress("Update failed! %s" % result.error_message())
+		yield(get_tree().create_timer(3), "timeout")
+		stop_progress()
```

Comment:
Would it work to re-enable the plugin here? That way the user isn't left with an unexpected disabled plugin.
Or would that re-launch the updater and make stuff confusing for the user?

Reply:
good catch